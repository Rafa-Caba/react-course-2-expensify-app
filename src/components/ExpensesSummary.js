import React from "react"
import { connect } from "react-redux"
import numeral from "numeral"
import selectExpenses from "../selectors/expenses"
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => (
    <div>
        { 
            <p>
                Viewing {expensesCount} {expensesCount === 1 ? 'expense' : 'expenses'} totalling {numeral(expensesTotal / 100).format('$0,00.00')}
            </p>
        }  
    </div>
)

const mapStateProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)

    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateProps)(ExpensesSummary)