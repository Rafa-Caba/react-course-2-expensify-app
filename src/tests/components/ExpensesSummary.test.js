import React from "react"
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should render one ExpenseSummary the total of the expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('should render one ExpenseSummary the total of the multi expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={25} expensesTotal={12345612346} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})
