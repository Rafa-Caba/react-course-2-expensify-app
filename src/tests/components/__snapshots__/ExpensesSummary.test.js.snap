// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render one ExpenseSummary the total of the expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </p>
</div>
`;

exports[`should render one ExpenseSummary the total of the multi expenses 1`] = `
<div>
  <p>
    Viewing 
    25
     
    expenses
     totalling 
    $123,456,123.46
  </p>
</div>
`;
